tests/unit/assessment/test_jobs.py

```diff
@@ -684,3 +695,77 @@ def test_job_crawler_with_no_owner_should_have_empty_creator_name():
     result = mockbackend.rows_written_for("hive_metastore.ucx.jobs", "append")
     assert result == [JobInfo(job_id="536591785949415", job_name="Unknown", creator=None, success=1, failures="[]")]
+
+
+@pytest.mark.parametrize(
+    "start_dttm, days_history",
+    [
+        (datetime(2021, 1, 1, 0, 0, 0), 1),
+        (datetime(2017, 1, 1, 0, 0, 0), 7),
+        (datetime(2025, 1, 1, 0, 0, 0), 30),
+    ],
+    ids=["1_day", "7_days", "30_days"],
+)
+def test_submit_runs_crawler(start_dttm: datetime, days_history: int):
+    ws = create_autospec(WorkspaceClient)
+    days_history = 1
+
+    with patch.object(SubmitRunsCrawler, "_get_current_dttm", return_value=start_dttm):
+        with patch.object(SubmitRunsCrawler, "_assess_job_runs", return_value=[]):
+            crawler = SubmitRunsCrawler(ws, MockBackend(), "ucx", days_history)
+            crawler._crawl()
+            time_from = SubmitRunsCrawler._dt_to_ms(start_dttm - timedelta(days=days_history))
+            time_to = SubmitRunsCrawler._dt_to_ms(start_dttm)
+
+            ws.jobs.list_runs.assert_called_once_with(
+                expand_tasks=True,
+                completed_only=True,
+                run_type=ListRunsRunType.SUBMIT_RUN,
+                start_time_to=time_to,
+                start_time_from=time_from,
+            )
+
+
+def test_dttm():
+    # cut off microseconds to avoid flakiness
+    assert datetime.now().replace(microsecond=0) == SubmitRunsCrawler._get_current_dttm().replace(microsecond=0)
+
+
+@pytest.mark.parametrize(
+    "spec, expected",
+    [
+        (ClusterSpec(spark_version="14.1.x-gpu-ml-scala2.12"), True),
+        (ClusterSpec(spark_version="10.1.x-scala2.12"), False),
+        (ClusterSpec(spark_version="9.1.x-scala2.12"), False),
+        (ClusterSpec(spark_version="11.1.x-scala2.12"), False),
+        (ClusterSpec(spark_version="11.2.x-scala2.12"), False),
+        (ClusterSpec(spark_version="11.3.x-scala2.12"), True),
+        (ClusterSpec(spark_version="11.3.x-scala2.12", data_security_mode=DataSecurityMode.NONE), False),
+        (ClusterSpec(spark_version="11.3.x-scala2.12", data_security_mode=DataSecurityMode.SINGLE_USER), False),
+        (ClusterSpec(spark_version="10.3.x-scala2.12", data_security_mode=DataSecurityMode.SINGLE_USER), False),
+    ],
+)
+def test_compat_check(spec, expected):
+    assert expected == SubmitRunsCrawler._needs_compatibility_check(spec)
+
+
+def test_submit_runs_assessment_empty():
+    crawler = SubmitRunsCrawler(Mock(), MockBackend(), "ucx", 10)
+    assert [] == crawler._assess_job_runs([])
+
+
+def test_submit_run_hashing_git_source_on_run_level():
+    r1 = BaseRun(
+        run_id=123,
+        git_source=GitSource(git_provider=GitProvider.GIT_HUB, git_url="http://some.git.repo"),
+    )
+
+    r2 = BaseRun(
+        run_id=123,
+        git_source=GitSource(git_provider=GitProvider.GIT_HUB, git_url="http://some.other.git.repo"),
+    )
+    h1 = SubmitRunsCrawler._get_hash_from_run(r1)
```

Comment:
accessing private members is prohibited

Reply:
Literally in the same file, line 118:
```
    result_set = JobsCrawler(Mock(), MockBackend(), "ucx")._assess_jobs(
        sample_jobs, {c.cluster_id: c for c in sample_clusters}
    )
```

A private methods is accessed. Do I understand correctly that it's prohibited personally for me and this PR, for others it's allowed?

Reply:
No, wait a second. Whichever principle is there, I see that my PR is intentionally reviewed way stricter than any other. Could you please provide any feasible explanation why is that?